### PR TITLE
coffee option fix, dry code, cleaner GF output, JS watch

### LINKED ISSUE #200: JS files are created with coffee option
app/scripts/main.js
test/spec/test.js

### PR DESCRIPTION
**Pull request by eddiemonge**
- Don't create JS files when --coffee is chosen
- JSHint and livereload JS in serve mode
- DRY up index.js a bit
- Get rid of trailing whitespace in generated Gruntfile.js

Fixes #200
BREAKING CHANGE: When choosing --coffee, JS files will no longer be
created automatically as well.

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -18,8 +18,6 @@ var AppGenerator = module.exports = function Appgenerator(args, options, config)
   // resolved to mocha by default (could be switched to jasmine for instance)
   this.hookFor('test-framework', { as: 'app' });
 
-  this.mainCoffeeFile = 'console.log "\'Allo from CoffeeScript!"';
```
Comment (eddiemonge) [inline app/index.js:21]: moved with the JS creation below

File: app/index.js [hunk 2]
```diff
@@ -104,14 +102,12 @@ AppGenerator.prototype.h5bp = function h5bp() {
 };
 
 AppGenerator.prototype.mainStylesheet = function mainStylesheet() {
-  if (this.compassBootstrap) {
-    this.copy('main.scss', 'app/styles/main.scss');
-  } else {
-    this.copy('main.css', 'app/styles/main.css');
-  }
+  var css = 'main.' + (this.compassBootstrap ? 's' : '') + 'css';
+  this.copy(css, 'app/styles/' + css);
```
Comment (eddiemonge) [inline app/index.js:106]: DRY'ed

File: app/index.js [hunk 3]
```diff
@@ -119,31 +115,22 @@ AppGenerator.prototype.writeIndex = function writeIndex() {
     'scripts/main.js'
   ]);
 
-  if (this.coffee) {
```
Comment (eddiemonge) [inline app/index.js:122]: Doesn't need a JS and a CS block when --coffee is chosen

File: app/index.js [hunk 4]
```diff
@@ -119,31 +115,22 @@ AppGenerator.prototype.writeIndex = function writeIndex() {
...
+      bs + 'dropdown.js',
+      bs + 'tooltip.js',
+      bs + 'modal.js',
+      bs + 'transition.js',
+      bs + 'button.js',
+      bs + 'popover.js',
+      bs + 'carousel.js',
+      bs + 'scrollspy.js',
+      bs + 'collapse.js',
+      bs + 'tab.js'
```
Comment (eddiemonge) [inline app/index.js:133]: DRY'ed

File: app/index.js [hunk 5]
```diff
@@ -156,8 +143,12 @@ AppGenerator.prototype.app = function app() {
...
 
   if (this.coffee) {
-    this.write('app/scripts/hello.coffee', this.mainCoffeeFile);
+    this.write(
+      'app/scripts/main.coffee',
+      'console.log "\'Allo from CoffeeScript!"'
+    );
+  }
+  else {
+    this.write('app/scripts/main.js', 'console.log(\'\\\'Allo \\\'Allo!\');');
```
Comment (eddiemonge) [inline app/index.js:152]: Creation is done in one place with the correct file created, not both

File: app/templates/Gruntfile.js [hunk 6]
```diff
@@ -35,6 +35,13 @@ module.exports = function (grunt) {
                 files: ['test/spec/{,*/}*.{coffee,litcoffee,coffee.md}'],
                 tasks: ['coffee:test', 'test:watch']
             },<% } else { %>
+            js: {
+                files: ['<%%= yeoman.app %>/scripts/{,*/}*.js'],
+                tasks: ['jshint'],
+                options: {
+                    livereload: true
+                }
+            },
```
Comment (eddiemonge) [inline app/templates/Gruntfile.js:44]: Turn on JSHint when JS files are saved and then reload the page

File: app/templates/Gruntfile.js [hunk 7]
```diff
@@ -56,8 +63,8 @@ module.exports = function (grunt) {
                 },
                 files: [
                     '<%%= yeoman.app %>/*.html',
-                    '.tmp/styles/{,*/}*.css',
-                    '{.tmp,<%%= yeoman.app %>}/scripts/{,*/}*.js',
+                    '.tmp/styles/{,*/}*.css',<% if (coffee) { %>
+                    '.tmp/scripts/{,*/}*.js',<% } %>
```
Comment (eddiemonge) [inline app/templates/Gruntfile.js:67]: only reload the page when CS is selected and the CS files are edited. JS in app is handled above

File: app/templates/Gruntfile.js [hunk 8]
```diff
@@ -128,7 +135,7 @@ module.exports = function (grunt) {
             ]
         },
 
-        <% if (testFramework === 'mocha') { %>
+<% if (testFramework === 'mocha') { %>
```
Comment (eddiemonge) [inline app/templates/Gruntfile.js:138]: these next three changes make the source file ugly but make the dest file pretty

**Comment by addyosmani** [comment]
Reviewed and LGTM. Thanks again, @eddiemonge!

**Pull request merged**